script.js

```diff
@@ -1,4 +1,593 @@
+      cardDeck.push(card);
+      cardCounter += 1;
+    }
+    suitIndex += 1;
+  }
+  return cardDeck;
+};
+
+//FUNCTION 2 - Shuffle deck.
+// Get a random index ranging from 0 (inclusive) to max (exclusive).
+var getRandomIndex = function (max) {
+  return Math.floor(Math.random() * max);
+};
+
+// Shuffle the elements in the cardDeck array
+var shuffleCards = function (cardDeck) {
+  var currentIndex = 0;
+  while (currentIndex < cardDeck.length) {
+    var randomIndex = getRandomIndex(cardDeck.length);
+    var randomCard = cardDeck[randomIndex];
+    var currentCard = cardDeck[currentIndex];
+    cardDeck[currentIndex] = randomCard;
+    cardDeck[randomIndex] = currentCard;
+    currentIndex = currentIndex + 1;
+  }
+  return cardDeck;
+};
+
+//FUNCTION 3a - DRAW HAND. i.e. Pop 2 initial cards from deck and store in array of playercard or computercard (whichever is applicable).
+var drawHand = function (shuffledDeck, cardArray) {
+  var initialCard1 = shuffledDeck.pop();
+  var initialCard2 = shuffledDeck.pop();
+  cardArray.push(initialCard1);
+  cardArray.push(initialCard2);
+
+  return cardArray;
+};
+
+//FUNCTION 3b - DRAW CARD. i.e. pop 1 card every time either player or computer has to HIT (draw a card). Then store into respective array.
+var drawCard = function (shuffledDeck, cardArray) {
+  var cardDrawn = shuffledDeck.pop();
+  cardArray.push(cardDrawn);
+
+  return cardArray;
+};
+
+//FUNCTION 4 -Calculate score of current array.
+//FUNCTION 4A - Create function to compare ranks of objects in array, with largest put before smallest.
+var sortFromLargest = function compare(a, b) {
+  if (a.rank > b.rank) {
+    return -1;
+  }
+  if (a.rank < b.rank) {
+    return 1;
+  }
+  return 0;
+};
+
+//FUNCTION 4B - Create function to sort current array from largest to smallest.
+//Create outer loop to sum up each card rank of card array as their default values. Ace = 11.
+//Store summed up value as initial score variable.
+//If initial score sum is <21, exit outer loop, return initial score.
+
+//If initial score sum is >21, create inner loop to detect if there is ace (or aces) within the card array.
+//If there is ace within card array, minus 10 to give ace a value of 1. Store as final score. See if final score <=21. If still more, repeat loop.
+//Until there is no more ace left and final score sum is still more than 21, OR the final score sum is less than 21.
+//Exit inner loop, return final score. Exit outer loop.
+
+//If initial score sum is >21, but there is no ace within card array, exit inner loop, make final score = initial score. Return final score. Exit outer loop.
+var calculateScore = function (cardArray) {
+  var sortedCardArray = cardArray.sort(sortFromLargest);
+  var currentInitialScore = 0;
+  var currentFinalScore = 0;
+  console.log(sortedCardArray);
+  var i = 0;
+  while (i < sortedCardArray.length) {
+    currentInitialScore += sortedCardArray[i].rank;
+    i += 1;
+  }
+  console.log(currentInitialScore);
```

Comment:
just fyi, during bootcamp and when you write code in production, once you're ready to commit something as "done", you should remove all the console.logs, cause these are for tests